sass/sass.bzl

```diff
@@ -11,151 +11,167 @@
-            single_file = True,
-        ),
-    },
-    outputs = {
-        "css_file": "%{name}.css",
-        "css_map_file": "%{name}.css.map",
-    },
-    implementation = _sass_binary_impl,
-)
-
-LIBSASS_BUILD_FILE = """
-package(default_visibility = ["@sassc//:__pkg__"])
-
-filegroup(
-    name = "srcs",
-    srcs = glob([
-         "src/**/*.h*",
-         "src/**/*.c*",
-    ]),
-)
-
-# Includes directive may seem unnecessary here, but its needed for the weird
-# interplay between libsass/sassc projects. This is intentional.
-cc_library(
-    name = "headers",
-    includes = ["include"],
-    hdrs = glob(["include/**/*.h"]),
 )
+"""Defines a group of Sass include files.
 """
 
-SASSC_BUILD_FILE = """
-package(default_visibility = ["//visibility:public"])
-
-cc_binary(
-    name = "sassc",
-    srcs = [
-        "@libsass//:srcs",
-        "sassc.c",
-        "sassc_version.h",
-    ] + select({
-        "@bazel_tools//src/conditions:windows": glob([
-            "win/**/*.c",
-            "win/**/*.h",
-        ]),
-        "//conditions:default": [],
-    }),
-    includes = select({
-        "@bazel_tools//src/conditions:windows": ["win/posix"],
-        "//conditions:default": [],
-    }),
-    linkopts = select({
-        "@bazel_tools//src/conditions:windows": ["-DEFAULTLIB:shell32.lib"],
-        "//conditions:default": [
-            "-ldl",
-            "-lm",
+_sass_binary_attrs = {
+    "src": attr.label(
+        doc = "Sass entrypoint file",
+        allow_files = _FILETYPES,
+        mandatory = True,
+        single_file = True,
+    ),
+    "include_paths": attr.string_list(
+        doc = "Additional directories to search when resolving imports"),
+    "output_dir": attr.string(
+        doc = "Output directory, relative to this package.",
+        default = ""),
+    "output_name": attr.string(
+        doc = """Name of the output file, including the .css extension.
+
+By default, this is based on the `src` attribute: if `styles.scss` is
+the `src` then the output file is `styles.css.`.
+You can override this to be any other name.
+Note that some tooling may assume that the output name is derived from
+the input name, so use this attribute with caution.""",
+        default = ""),
+    "output_style": attr.string(
+        doc = "How to style the compiled CSS",
+        default = "compressed",
+        values = [
+            "expanded",
+            "compressed",
         ],
-    }),
-    deps = ["@libsass//:headers"],
-)
-"""
-
-def sass_repositories():
-  native.new_http_archive(
-      name = "libsass",
-      url = "http://bazel-mirror.storage.googleapis.com/github.com/sass/libsass/archive/3.3.0-beta1.tar.gz",
-      sha256 = "6a4da39cc0b585f7a6ee660dc522916f0f417c890c3c0ac7ebbf6a85a16d220f",
-      build_file_content = LIBSASS_BUILD_FILE,
-      strip_prefix = "libsass-3.3.0-beta1",
-  )
+    ),
+    "deps": sass_deps_attr,
+    "compiler": attr.label(
+        doc = """Choose which Sass compiler binary to use.
```

Comment:
Is there a way to break large docs like this one out somewhere else? It rather breaks up the code-reading as it is now.